Lib/logging/__init__.py

```diff
@@ -1543,9 +1558,17 @@ def isEnabledFor(self, level):
         """
         Is this logger enabled for level 'level'?
         """
-        if self.manager.disable >= level:
-            return False
-        return level >= self.getEffectiveLevel()
+        try:
+            return self._cache[level]
+        except KeyError:
+            _acquireLock()
+            if self.manager.disable >= level:
```

Comment:
Would it not make more sense to cache the result of `getEffectiveLevel()`, rather than `isEnabledFor()`? From what I can see, the time saving is really from cutting out the hierarchy walk in the former method.